spring-web/src/main/java/org/springframework/web/client/DefaultRestClient.java

```diff
@@ -452,6 +458,9 @@ private <T> T exchangeInternal(ExchangeFunction<T> exchangeFunction, boolean clo
 			URI uri = null;
 			try {
 				uri = initUri();
+				if (defaultRequest != null) {
+					defaultRequest.accept(this);
+				}
```

Comment:
https://github.com/spring-projects/spring-framework/blob/faba044735ad173d96b4b1adab62752b48b2c758/spring-webflux/src/main/java/org/springframework/web/reactive/function/client/DefaultWebClient.java#L481-L484

FYI) almost same usage of `defaultRequest` on `WebClient`~!